File: android-client/app/src/main/java/com/example/stripedemo/data/models/AccountLink.kt
package com.example.stripedemo.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class AccountLinkRequest(
    @Json(name = "refresh_url")
    val refreshUrl: String,
    @Json(name = "return_url")
    val returnUrl: String
)

@JsonClass(generateAdapter = true)
data class AccountLinkResponse(
    val url: String,
    val created: String,
    @Json(name = "expires_at")
    val expiresAt: String
)

File: ios-client/StripeApplication/StripeApplication/Data/Models/AccountLink.swift
import Foundation

struct AccountLinkRequest: Codable {
    let refreshUrl: String
    let returnUrl: String

    enum CodingKeys: String, CodingKey {
        case refreshUrl = "refresh_url"
        case returnUrl = "return_url"
    }
}

struct AccountLinkResponse: Codable {
    let url: String
    let created: String
    let expiresAt: String

    enum CodingKeys: String, CodingKey {
        case url, created
        case expiresAt = "expires_at"
    }
}
